deltachat-ios/Controller/ConnectivityViewController.swift

```diff
@@ -41,10 +41,88 @@ class ConnectivityViewController: WebViewViewController {
         }
     }
 
+    // this method needs to be run from a background thread
+    private func getNotificationStatus() -> String {
+        let title = " <b>" + String.localized("pref_notifications") + ":</b> "
+        let notificationsEnabledInDC = !UserDefaults.standard.bool(forKey: "notifications_disabled")
+        var notificationsEnabledInSystem = false
+        let semaphore = DispatchSemaphore(value: 0)
+        DispatchQueue.global(qos: .userInitiated).async {
+            NotificationManager.notificationEnabledInSystem { enabled in
+                notificationsEnabledInSystem = enabled
+                semaphore.signal()
+            }
+        }
+        if semaphore.wait(timeout: .now() + 1) == .timedOut {
+            return "<span class=\"red dot\"></span>"
+                .appending(title)
+                .appending("Timeout Error")
+        }
+
+        if !notificationsEnabledInDC {
+            return "<span class=\"disabled dot\"></span>"
+                .appending(title)
+                .appending(String.localized("notifications_disabled_dc"))
+        }
+
+        if !notificationsEnabledInSystem {
+            return "<span class=\"disabled dot\"></span>"
+                .appending(title)
+                .appending(String.localized("notifications_disabled"))
+        }
+
+        let timestamps = UserDefaults.standard.array(forKey: Constants.Keys.notificationTimestamps) as? [Double]
+        guard let timestamps = timestamps else {
+            return "<span class=\"yellow dot\"></span>"
+                .appending(title)
+                .appending(String.localized("no_data"))
+        }
+
+        if timestamps.isEmpty || timestamps.count == 1 {
+            // FIXME: for timestamp == 1, that is just okay if the timestamp is not too old
+            return "<span class=\"red dot\"></span>"
+                .appending(title)
+                .appending(String.localized("notifications_not_working"))
```

Comment:
this has the issue, that if there is the first wakeup coming, this results in "no working" display.

with the second one, things are fine.

i think, maybe we should just calculate the average by `(now - oldest) / count` - that would require only one data point to work.